pxf/pxf-hive/src/main/java/org/apache/hawq/pxf/plugins/hive/utilities/HiveUtilities.java

```diff
@@ -203,19 +205,37 @@ private static boolean verifyModifers(String[] modifiers) {
      * It can be either <code>table_name</code> or <code>db_name.table_name</code>.
      *
      * @param qualifiedName Hive table name
-     * @return {@link org.apache.hawq.pxf.api.Metadata.Table} object holding the full table name
+     * @return {@link Metadata.Item} object holding the full table name
      */
-    public static Metadata.Table parseTableQualifiedName(String qualifiedName) {
+    public static Metadata.Item extractTableFromName(String qualifiedName) {
+        List<Metadata.Item> items = extractTablesFromPattern(null, qualifiedName);
+        if(items.isEmpty()) {
+            throw new IllegalArgumentException("No tables found");
```

Comment:
It's hard to parse stacktrace on client-side, so we should serialize all errors to JSON.